Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,714 @@
+  Function.Injective.smulWithZero ⟨_, coe_zero⟩ DFunLike.coe_injective coe_smul
+
+instance instMulActionWithZero [Zero β] {R : Type*} [MonoidWithZero R] [MulActionWithZero R β]
+    [ContinuousConstSMul R β] : MulActionWithZero R C_c(α, β) :=
+  Function.Injective.mulActionWithZero ⟨_, coe_zero⟩ DFunLike.coe_injective coe_smul
+
+instance instModule [AddCommMonoid β] [ContinuousAdd β] {R : Type*} [Semiring R] [Module R β]
+    [ContinuousConstSMul R β] : Module R C_c(α, β) :=
+  Function.Injective.module R ⟨⟨_, coe_zero⟩, coe_add⟩ DFunLike.coe_injective coe_smul
+
+instance instNonUnitalNonAssocSemiring [NonUnitalNonAssocSemiring β] [TopologicalSemiring β] :
+    NonUnitalNonAssocSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance instNonUnitalSemiring [NonUnitalSemiring β] [TopologicalSemiring β] :
+    NonUnitalSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance instNonUnitalCommSemiring [NonUnitalCommSemiring β] [TopologicalSemiring β] :
+    NonUnitalCommSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance instNonUnitalNonAssocRing [NonUnitalNonAssocRing β] [TopologicalRing β] :
+    NonUnitalNonAssocRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance instNonUnitalRing [NonUnitalRing β] [TopologicalRing β] : NonUnitalRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalRing _ coe_zero coe_add coe_mul coe_neg coe_sub (fun _ _ => rfl)
+    fun _ _ => rfl
+
+instance instNonUnitalCommRing [NonUnitalCommRing β] [TopologicalRing β] :
+    NonUnitalCommRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance instIsScalarTower {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [IsScalarTower R β β] :
+    IsScalarTower R C_c(α, β) C_c(α, β) where
+  smul_assoc r f g := by
+    ext
+    simp only [smul_eq_mul, coe_mul, coe_smul, Pi.mul_apply, Pi.smul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_assoc]
+
+instance instSMulCommClass {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [SMulCommClass R β β] :
+    SMulCommClass R C_c(α, β) C_c(α, β) where
+  smul_comm r f g := by
+    ext
+    simp only [smul_eq_mul, coe_smul, coe_mul, Pi.smul_apply, Pi.mul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_comm]
+
+end AlgebraicStructure
+
+section ZeroAtInfty
+
+open ZeroAtInfty
+
+variable [TopologicalSpace β] [TopologicalSpace γ] [Zero γ]
+variable [FunLike F β γ] [CompactlySupportedContinuousMapClass F β γ]
+
+lemma zero_at_infty_of_hasCompactSupport (f : F) :
```

Comment:
The lemma name is not very good, since `hasCompactSupport` does not appear in the name. Instead, it should be `zero_at_infty_of_compactlySupportedContinuousMapClass`. But it's probably worth it to have a standalone lemma, in another file, saying that a function with compact support tends to zero at infinity -- and to use this lemma here to get a very short proof.

In fact, we already have this lemma, `HasCompactSupport.is_zero_at_infty`, so you should just use it to shorten your proof.

Reply:
[HasCompactSupport.is_zero_at_infty](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/UniformSpace/Compact.html#HasCompactSupport.is_zero_at_infty) assumes` [UniformSpace α]`. I guess this assumption is unnecessary? I wrote a lemma in `Topology/Support`. What should I do with the file `Topology.UniformSpace.Compact`?

Reply:
Yes, the current assumption if `UniformSpace` is not necessary at all, and the lemma is not in the right file. You should move the lemma to the file `Topology/Support`, but not change its name (and not change its proof, because it already works perfectly fine -- you can see that your version of the proof is a little less efficient).

Reply:
I did it, except that I did not copy the `#align` lines. I hope that is fine.

Reply:
sorry, I didn't know that it needs to be there after being moved. I restored it.